sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/NumberPatternShouldBeRegular.cs

```diff
@@ -73,5 +75,21 @@ protected override void Initialize(SonarAnalysisContext context)
                 },
                 SyntaxKind.NumericLiteralExpression);
         }
+
+        private static string ClearNumberTypeSuffix(string numberLiteral)
+        {
+            if (numberLiteral.EndsWith("UL"))
```

Comment:
the string checks should be case insensitive (in the if below too)
- integers - https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#integer-literals
- real numbers - https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#real-literals